include/battle_gimmick.h

```diff
@@ -0,0 +1,51 @@
+#ifndef GUARD_BATTLE_GIMMICK_H
+#define GUARD_BATTLE_GIMMICK_H
+
+enum Gimmick
+{
+    GIMMICK_NONE,
+    GIMMICK_MEGA,
+    GIMMICK_ULTRA_BURST,
+    GIMMICK_Z_MOVE,
+    GIMMICK_DYNAMAX,
+    GIMMICK_TERA,
+    GIMMICKS_COUNT,
+};
+
+struct GimmickInfo
+{
+    const struct SpritePalette * triggerPal;        // trigger gfx data
+    const struct SpriteSheet * triggerSheet;
+    const struct SpriteTemplate * triggerTemplate;
+    const struct SpritePalette * indicatorPal;      // indicator gfx data
+    const struct SpriteSheet * indicatorSheet;
+    bool32 (*CanActivate)(u32 battler);
+    void (*ActivateGimmick)(u32 battler);
+};
+
+void AssignUsableGimmicks(void);
+bool32 CanActivateGimmick(u32 battler, enum Gimmick gimmick);
```

Comment:
I actually can't recall any other functions where arguments are specified to need to come from an enum, but it's a fascinating thing that I think could be used way more across the codebase (also in pret).